src/GameData.ts

```diff
@@ -0,0 +1,320 @@
+        const champions = await fetch(this.sharedSettings.lookup.championUrl).then(x => x.json());
+
+        const returnData: ChampionData[] = champions
+            .filter((c: any) => c.id > 0)
+            .map((c: any) => ({
+                id: c.id,
+                name: c.name,
+                key: c.alias,
+                skins: [],
+            }));
+
+        const skins = await this.loadSkinData();
+        for (const champion of returnData) {
+            champion.skins = skins.filter(x => Math.floor(x.id / 1000) === champion.id);
+        }
+
+        return returnData;
+    }
+
+    public async loadSkinData(): Promise<SkinData[]> {
+        return Object.values(await fetch(this.sharedSettings.lookup.skinUrl).then(x => x.json())) as SkinData[];
+    }
+
+    public async loadPerkData(): Promise<PerkData[]> {
+        return await fetch(this.sharedSettings.lookup.perkUrl).then(x => x.json());
+    }
+
+    public async loadItemData(): Promise<ItemData[]> {
+        const items = await fetch(this.sharedSettings.lookup.itemUrl).then(x => x.json()) as ItemData[];
+        for (const item of items) {
+            const other = items.find(x => x.id === item.id)!;
+
+            other.from = item.from.map(x => items.find(y => y.id === +x)!).filter(x => x).map(x => x.name);
+            other.to = item.to.map(x => items.find(y => y.id === +x)!).filter(x => x).map(x => x.name);
+        }
+
+        return items;
+    }
+
+    public onLookup(message: Discord.Message, isAdmin: boolean, command: string, args: string[]) {
+        const supportedTypes = ["item", "perk", "rune", "champion"];
+
+        if (args.length === 0) {
+            const response = `Usage: !${command} [type] [term]. Supported types are ` + supportedTypes.map(x => "`" + x + "`").join(", ");
+            message.channel.send(response);
+            return;
+        }
+
+        if (!supportedTypes.includes(args[0])) {
+            message.channel.send(`I'm sorry. I'm unable to parse the category \`${args[0]}\` at this moment. If you think it should be added, please contact a guru.`);
+            return;
+        }
+
+        const searchTerm = args.slice(1).join(" ").toLowerCase();
+        let result: string | any[] = [];
+        switch (args[0]) {
+            case "item": {
+                result = this.findItem(searchTerm);
+                break;
+            }
+            case "perk": // fall through
+            case "rune": {
+                result = this.findPerk(searchTerm);
+                break;
+            }
+            case "champion": {
+                result = this.findChampion(searchTerm);
+                break;
+            }
+        }
+
+        if (typeof result === "string") {
+            message.channel.send(result);
+            return;
+        }
+
+        result.forEach(x => message.channel.send("```" + JSON.stringify(x, (k, v) => { if (v !== null) return v; }, 4) + "```"));
```

Comment:
`(k, v) => v !== null ? v : undefined` (or `void 0`)

Reply:
Also add a comment saying that you're using this to remove keys with `null` as value from the json representation. The second argument is rarely used so people may not know what it does.